verifier/src/lib.rs

```diff
@@ -51,114 +60,122 @@ impl<E: DisplayableError> VerificationResult<E> {
     /// representation with.
     pub fn fmt_padded(&self, f: &mut Formatter, pad: usize) -> core::fmt::Result {
         if self.is_ok.into() {
-            write!(f, "{:pad$}Passed", "")?;
+            self.metadata.fmt_success(f, pad)?;
         } else {
-            self.error.fmt_padded(f, pad)?;
+            self.metadata.fmt_failure(f, pad)?;
         }
         Ok(())
     }
 }
 
-/// Number of spaces to indent nested error messages.
-const ERROR_INDENT: usize = 2;
+impl<T: ResultMessage> Display for VerificationResult<T> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        self.fmt_padded(f, 0)
+    }
+}
 
-/// An error that implements the [`Display`] trait.
-pub trait DisplayableError: Display + Clone {
-    /// Format the error with preceding padding
+/// A message that can be used in [`VerificationResult`]'s [`Display`]
+/// implementation.
+pub trait ResultMessage {
+    /// Formatting of a successful verification with this message type
     ///
-    /// The `pad` is the number of spaces to precede each line of the error
+    /// The `pad` is the number of spaces to precede each line of the message
     /// with.
-    fn fmt_padded(&self, f: &mut Formatter, pad: usize) -> core::fmt::Result {
-        write!(f, "{:pad$}{self}", "")
-    }
+    fn fmt_success(&self, f: &mut Formatter, pad: usize) -> core::fmt::Result;
+
+    /// Formatting of a failure verification with this message type
+    ///
+    /// The `pad` is the number of spaces to precede each line of the message
+    /// with.
+    fn fmt_failure(&self, f: &mut Formatter, pad: usize) -> core::fmt::Result;
```

Comment:
Explicitly asking for the success vs failure seemed like the clearest way to be able to print all the info in a meaningful way to to the caller.